configuration/src/main/java/io/camunda/configuration/beanoverrides/BrokerBasedPropertiesOverride.java

```diff
@@ -795,112 +785,146 @@ private void populateCamundaExporter(final BrokerBasedProperties override) {
       exporter.setArgs(args);
     }
 
-    setArg(args, "history.retention.enabled", secondaryStorage.getRetention().isEnabled());
-    setArg(args, "history.retention.minimumAge", secondaryStorage.getRetention().getMinimumAge());
-
-    setArg(args, "connect.type", secondaryStorage.getType().name());
-    setArg(args, "connect.url", database.getUrl());
-    setArg(args, "connect.clusterName", database.getClusterName());
-    setArg(args, "connect.dateFormat", database.getDateFormat());
-    final var socketTimeout = database.getSocketTimeout();
-    setArg(args, "connect.socketTimeout", socketTimeout != null ? socketTimeout.toMillis() : null);
-    final var connectionTimeout = database.getConnectionTimeout();
-    setArg(
-        args,
-        "connect.connectTimeout",
-        connectionTimeout != null ? connectionTimeout.toMillis() : null);
+    exporter.setArgs(
+        ExporterConfiguration.of(io.camunda.exporter.config.ExporterConfiguration.class, args)
+            .apply(config -> applyCamundaExporterConfiguration(config, secondaryStorage, database))
+            .toArgs());
+  }
+
+  private void applyCamundaExporterConfiguration(
+      final io.camunda.exporter.config.ExporterConfiguration config,
+      final SecondaryStorage secondaryStorage,
+      final DocumentBasedSecondaryStorageDatabase database) {
 
-    // Add security configuration mapping
-    if (database.getSecurity() != null) {
-      setArg(args, "connect.security.enabled", database.getSecurity().isEnabled());
-      setArg(args, "connect.security.certificatePath", database.getSecurity().getCertificatePath());
-      setArg(args, "connect.security.verifyHostname", database.getSecurity().isVerifyHostname());
-      setArg(args, "connect.security.selfSigned", database.getSecurity().isSelfSigned());
+    applyCamundaExporterConfigurationRetention(
+        config.getHistory().getRetention(), secondaryStorage);
+    applyCamundaConfigurationConnect(config.getConnect(), secondaryStorage, database);
```

Comment:
🔧 this is all a bit much that we hand in and it also overlaps 🤔 how about this:
1. we hand in the whole `config` object to each of these helper methods
2. first LoC in each of these helper methods pulls out the needed sub-configs (e.g. secondary storage)
3. we add a `getResolvedDatabase` helper to the secondaryStorage config, we have already done the validation in the calling method that in this case it can only be ES/OS (thus it's safe to call)

wdyt?